helm/drone-fork-approval-extension/templates/deployment.yaml

```diff
@@ -0,0 +1,57 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "drone-fork-approval-extension.fullname" . }}
+  labels:
+    {{- include "drone-fork-approval-extension.labels" . | nindent 4 }}
+spec:
+  replicas: {{ .Values.replicaCount }}
+  selector:
+    matchLabels:
+      {{- include "drone-fork-approval-extension.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      {{- with .Values.podAnnotations }}
+      annotations:
+        checksum/secret: {{ .Values.secret | sha256sum }}
```

Comment:
This should be OK as long as best practices are used to generate the secret, as in a long random number. If the secret gets generated closer to a password, there might be a theoretical brute force against a fast checksum.

IDK how important this is, since I don't know what the threat model is for the drone secret on the validation plugin.

Reply:
Currently the "far side" of this secret is handled as an unecrypted env value, so the use of Kuberenetes secrets was overkill on this end.

https://github.com/drone/charts/blob/498bf298c0000acafa47b43308f4d06c6ab40e66/charts/drone/values.yaml#L154-L158

However, now that I look closer, we probably need to clean up the far side to use https://github.com/drone/charts/blob/498bf298c0000acafa47b43308f4d06c6ab40e66/charts/drone/values.yaml#L151

Do you know of a way to have helm autogenerate a secret value?  This is what I really want, as the user shouldn't need to specify this, and we can be sure they get a nice long secret.

Also, I added this because without it the deployment wasn't rotating on secret update.  Is there a better way to achieve that goal?

Reply:
You can autogenerate values in Helm with `randAlphaNum` but it has a lot of caveats - see https://stackoverflow.com/a/56171690/11329621

I think the sanest solution is just to allow the user to provide their own secret, as at least then it's known and consistent. People's choice of secrets is out of the chart's purview.

I think a sha256 hash is fine from a brute force perspective.

Reply:
Sounds good.  I won't change this for now.  Let me know if the secret checksum is still a concern to either of you.